src/XCCDF/xccdf_session.c

```diff
@@ -1265,14 +1274,15 @@ static int _build_xccdf_result_source(struct xccdf_session *session)
 	}
 
 	/* Build oscap_source of XCCDF TestResult only when needed */
-	if (session->export.xccdf_file != NULL || session->export.report_file != NULL || session->export.arf_file != NULL) {
+	if (session->export.xccdf_file != NULL || session->export.report_file != NULL || session->export.arf_file != NULL || session->export.stig_rule_id_file != NULL) {
 		if (session->xccdf.result == NULL) {
 			// Attempt to export session before evaluation
 			oscap_seterr(OSCAP_EFAMILY_OSCAP, "No XCCDF results to export.");
 			return 1;
 		}
-		xccdf_benchmark_add_result(xccdf_policy_model_get_benchmark(session->xccdf.policy_model),
-				xccdf_result_clone(session->xccdf.result));
+
+		struct xccdf_result *benchmarks_result = xccdf_result_clone(session->xccdf.result);
+		xccdf_benchmark_add_result(xccdf_policy_model_get_benchmark(session->xccdf.policy_model),	benchmarks_result);
```

Comment:
code style, there is a tab as you pass the benchmark. it's also not clear why you made this change because you don't even use this variable anywhere